include/boost/math/tools/polynomial.hpp

```diff
@@ -572,45 +609,61 @@ inline polynomial<T> operator - (const polynomial<T>& a, const polynomial<T>& b)
    result -= b;
    return result;
 }
+#ifndef BOOST_NO_CXX11_RVALUE_REFERENCES
+template <class T>
+inline polynomial<T> operator - (polynomial<T>&& a, const polynomial<T>& b)
+{
+   a -= b;
+   return a;
+}
+template <class T>
+inline polynomial<T> operator - (const polynomial<T>& a, polynomial<T>&& b)
+{
+   b -= a;
+   return -b;
+}
+template <class T>
+inline polynomial<T> operator - (polynomial<T>&& a, polynomial<T>&& b)
+{
+   a -= b;
+   return a;
+}
+#endif
 
 template <class T>
 inline polynomial<T> operator * (const polynomial<T>& a, const polynomial<T>& b)
 {
-   polynomial<T> result(a);
-   result *= b;
+   polynomial<T> result;
+   result.multiply(a, b);
    return result;
 }
 
 template <class T, class U>
-inline polynomial<T> operator + (const polynomial<T>& a, const U& b)
+inline polynomial<T> operator + (polynomial<T> a, const U& b)
```

Comment:
@jzmaddock I am a little confused by your comment. Here is what I think:

1. This function takes a polynomial and scalar, adds them and returns the result as a new polynomial
2. Suppose that we keep the signature I suggested:
  * If the input is a lvalue, a copy is made due to passing by value, scalar is added (does not allocate any memory) and we return the polynomial i.e. 1 memory allocation
  * If the input is a rvalue, the move constructor is invoked when passing and no copy is made. In the function body, we add using += and return the polynomial. This involves no copying.
3. I don't think we can do any better in terms of memory allocation by adding separate overloads for lvalue and rvalue.

Please let me know your views on this.